controllers/sso_configuration.go

```diff
@@ -0,0 +1,62 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	repomanagerv1alpha1 "github.com/git-hyagi/pulp-operator-go/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// ssoConfig sets the configurations needed to authenticate pulp through keycloak
+func (r *PulpReconciler) ssoConfig(ctx context.Context, pulp *repomanagerv1alpha1.Pulp, pulpSettings *string) error {
+	log := ctrllog.FromContext(ctx)
+
+	// Check for specified sso configuration secret
+	secret := &corev1.Secret{}
+	err := r.Get(ctx, types.NamespacedName{Name: pulp.Spec.SSOSecret, Namespace: pulp.Namespace}, secret)
+	if err != nil {
+		log.Error(err, "Failed to find "+pulp.Spec.SSOSecret+" secret")
+		return err
+	}
```

Comment:
why not use [retrieveSecretData](https://github.com/git-hyagi/pulp-operator-go/blob/main/controllers/common.go#L37) ?

Reply:
Good idea!
Working on it